lib/publishing_api_finder_publisher.rb

```diff
@@ -3,25 +3,44 @@
 require_relative "../app/presenters/finder_signup_content_item_presenter"
 
 class PublishingApiFinderPublisher
-  def initialize(metadata, schemae)
+  def initialize(metadata, schemae, log = true)
     @metadata = metadata
     @schemae = schemae
+    @log = log
   end
 
   def call
-    metadata.zip(schemae).map { |metadata, schema|
-      if metadata[:file].has_key?("content_id")
-        export_finder(metadata, schema)
-        export_signup(metadata) if metadata[:file].has_key?("signup_content_id")
+    metadata.zip(schemae).map do |metadata, schema|
+      if metadata[:file].has_key?("content_id") && !preview_only?(metadata)
+        publish metadata, schema
+      elsif preview_only?(metadata)
+        if preview_domain_or_not_production?
+          publish metadata, schema
+        else
+          puts "didn't publish #{metadata[:file]["name"]} because it is preview_only" if @log
+        end
       else
-        puts "didn't publish #{metadata[:file]["name"]} because it doesn't have a content_id"
+        puts "didn't publish #{metadata[:file]["name"]} because it doesn't have a content_id" if @log
       end
-    }
+    end
   end
 
 private
   attr_reader :schemae, :metadata
 
+  def publish metadata, schema
+    export_finder(metadata, schema)
+    export_signup(metadata) if metadata[:file].has_key?("signup_content_id")
+  end
+
+  def preview_only? metadata
+    metadata[:file]["preview_only"] == true
+  end
+
+  def preview_domain_or_not_production?
+    ENV.fetch("GOVUK_APP_DOMAIN", "")[/preview/] || !Rails.env.production?
```

Comment:
This is the wrong way to do this as it'll break if we ever change host names. We should be using some config, eg an initializer, which is overridden in preview and development.

Something like `config.publish_preview_finders = true`.